lib/honeycomb/propagation/w3c.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+require "base64"
+require "json"
+require "uri"
+
+module Honeycomb
+  # Parsing and propagation for W3C trace headers
+  module W3CPropagation
+    # Parse trace headers
+    module UnmarshalTraceContext
+      def parse(serialized_trace)
+        unless serialized_trace.nil?
+          version, payload = serialized_trace.split("-", 2)
+          if version == "00"
+            trace_id, parent_span_id = parse_v1(payload)
+
+            if !trace_id.nil? && !parent_span_id.nil?
+              # return nil for dataset
+              return [trace_id, parent_span_id, nil, nil]
+            end
+          end
+        end
+        [nil, nil, nil, nil]
+      end
+
+      def parse_v1(payload)
+        @invalid_trace_id = "00000000000000000000000000000000"
+        @invalid_span_id = "0000000000000000"
```

Comment:
These could probably be constants in this file?

Reply:
yes, I think I went overboard with the `@` variables so I've removed those, but let me know if I'm interpreting this change incorrectly!